### ISSUE TITLE
ChangeLog and merge policy

### ISSUE THREAD

**Issue by chdoc** [EasyCrypt/easycrypt#193]
This is to discuss (and document) the ChangeLog an merge policy

@strub https://github.com/EasyCrypt/easycrypt/pull/190#issuecomment-1123760884

> Should we generate the ChangeLog from the history before the release? (We enforce a linear history now and should reject PR with unclear commit messages)

**Comment by strub** [comment]
> I think the plan is to squash-merge PRs, which means crafting a commit message anyway.

No. The plan is to rebase-merge, not even squash-merge. Squash-merge tends to hide the real author and the PR might have a sane history made of multiple commits. Merge commits make cherry-picking, bisecting, reverting harder.

Regarding the reference to the PR, you can search a PR by commit SHA.

**Comment by strub** [comment]
I don't have a strong opinion about the ChangeLog. Initially, I was thinking about enforcing a template for commit messages, s.t. we could extract a ChangeLog from them. But if people prefer to update the ChangeLog gradually, we can go for that. Here too, we should have some script that checks for the ChangeLog format.
